.gitlab-ci.yml

```diff
@@ -0,0 +1,68 @@
+
+stages:
+  - test
+  - publish
+
+test:unit:
+  image: golang:1.13
+  stage: test
+
+  before_script:
+    # Rename the branch we're on, so that it's not in the way for the
+    # subsequent fetch. It's ok if this fails, it just means we're not on any
+    # branch.
+    - git branch -m temp-branch || true
+    # Git trick: Fetch directly into our local branches instead of remote
+    # branches.
+    - git fetch origin 'refs/heads/*:refs/heads/*'
+    # Get last remaining tags, if any.
+    - git fetch --tags origin
+
+    # Prepare GO path
+    - mkdir -p /go/src/github.com/mendersoftware
+    - cp -r $CI_PROJECT_DIR /go/src/github.com/mendersoftware/log
+    - cd /go/src/github.com/mendersoftware/log
+
+    # Install code coverage / coveralls tooling
+    - go get -u github.com/axw/gocov/gocov
+    - go get -u golang.org/x/tools/cmd/cover
+
+    # Install cyclomatic dependency analysis tool
+    - go get -u github.com/fzipp/gocyclo
+
+  script:
+    # Test if code was formatted with 'go fmt'
+    # Command will format code and return modified files
+    # fail if any have been modified.
+    - if [ -n "$(go fmt)" ]; then echo 'Code is not formatted with "go fmt"'; false; fi
+
+    # Perform static code analysys
+    - go vet `go list ./... | grep -v vendor`
+
+    # Fail builds when the cyclomatic complexity reaches 15 or more
+    - gocyclo -over 15 `find . -iname '*.go' | grep -v 'vendor' | grep -v '_test.go'`
+
+    # go list supply import paths for all sub directories.
+    # Exclude vendor directory, we don't want to run tests and coverage for all dependencies every time,
+    # also including their coverage may introduce to much noice. Concentrate on the coverage of local packages.
+    # Execute go test on every local subpackage (resolved as dependencies) and generate covreage report for each.
+    # Test packages pararell (xargs -P)
+    - go list ./... | grep -v vendor | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {} || exit $?
+
+    # Collect coverage report
+    - cp coverage.txt $CI_PROJECT_DIR/
```

Comment:
We should collect all coverage.txt files. See https://github.com/mendersoftware/go-lib-micro/pull/52

Reply:
@lluiscampos 
I'm really sorry to disturb you, but can you help me to understand what's up with codecov.io now?
https://github.com/mendersoftware/log/pull/17
https://gitlab.com/Northern.tech/Mender/log/-/jobs/374534552

Reply:
The artifact collected at unit test job (can be found [here](https://gitlab.com/Northern.tech/Mender/log/-/jobs/374534551) -> Browse) is correct. The codecov script finds it and theoretically pushes it.

There seems to be a problem in the GitHub settings for this repo (remember that this repo was usind coveralls): there is no webhook here for codecov. I will try to fix and re-trigger the build.

Reply:
It works now. The only thing I did was delete the integration, give some time, add the integration, give more time, and retry.

Reply:
Here the report: https://codecov.io/gh/mendersoftware/log/commit/de4e4e65d7dd582a39b8e35b2f06a2123f04657f